datafusion/datasource-parquet/src/source.rs

```diff
@@ -349,11 +337,13 @@ impl ParquetSource {
     }
 
     /// Optional reference to this parquet scan's pruning predicate
+    #[deprecated(note = "ParquetDataSource no longer constructs a PruningPredicate.")]
     pub fn pruning_predicate(&self) -> Option<&Arc<PruningPredicate>> {
         self.pruning_predicate.as_ref()
```

Comment:
I suggest we change this code to always return None (and remove the page pruning predicate from the source).

Otherwise I predict that we'll end up with this field slowly bitrotting (aka stop working but not covered by tests)

```suggestion
        None
```

I sugget the same thing for page_pruning_predicate

Reply:
I wanted to do this as well but it broke several tests that rely on this. Are you okay with a larger diff to handle this? I was thinking we do it in a followup PR (I can make a ticket to track and deal with it right after this one is merged).

Reply:
a follow on ticket / PR sounds like a good idea to me

Reply:
For anyone following along

- https://github.com/apache/datafusion/issues/15534